powertools-idempotency/powertools-idempotency-redis/src/main/java/software/amazon/lambda/powertools/idempotency/redis/RedisPersistenceStore.java

```diff
@@ -0,0 +1,404 @@
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPooled;
+import redis.clients.jedis.UnifiedJedis;
+import software.amazon.lambda.powertools.idempotency.exceptions.IdempotencyItemAlreadyExistsException;
+import software.amazon.lambda.powertools.idempotency.exceptions.IdempotencyItemNotFoundException;
+import software.amazon.lambda.powertools.idempotency.persistence.BasePersistenceStore;
+import software.amazon.lambda.powertools.idempotency.persistence.DataRecord;
+import software.amazon.lambda.powertools.idempotency.persistence.PersistenceStore;
+
+/**
+ * Redis version of the {@link PersistenceStore}. Stores idempotency data in Redis standalone or cluster mode.<br>
+ * Use the {@link Builder} to create a new instance.
+ */
+public class RedisPersistenceStore extends BasePersistenceStore implements PersistenceStore {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RedisPersistenceStore.class);
+    private final String keyPrefixName;
+    private final String keyAttr;
+    private final String expiryAttr;
+    private final String inProgressExpiryAttr;
+    private final String statusAttr;
+    private final String dataAttr;
+    private final String validationAttr;
+    private final UnifiedJedis jedisClient;
+
+    /**
+     * Private: use the {@link Builder} to instantiate a new {@link RedisPersistenceStore}
+     */
+    private RedisPersistenceStore(String keyPrefixName,
+                                  String keyAttr,
+                                  String expiryAttr,
+                                  String inProgressExpiryAttr,
+                                  String statusAttr,
+                                  String dataAttr,
+                                  String validationAttr,
+                                  UnifiedJedis jedisClient) {
+        this.keyPrefixName = keyPrefixName;
+        this.keyAttr = keyAttr;
+        this.expiryAttr = expiryAttr;
+        this.inProgressExpiryAttr = inProgressExpiryAttr;
+        this.statusAttr = statusAttr;
+        this.dataAttr = dataAttr;
+        this.validationAttr = validationAttr;
+
+        if (jedisClient != null) {
+            this.jedisClient = jedisClient;
+        } else {
+            String idempotencyDisabledEnv =
+                    System.getenv(software.amazon.lambda.powertools.idempotency.Constants.IDEMPOTENCY_DISABLED_ENV);
+            if (idempotencyDisabledEnv == null || "false".equalsIgnoreCase(idempotencyDisabledEnv)) {
+                this.jedisClient = getJedisClient(System.getenv(Constants.REDIS_HOST), Integer.parseInt(System.getenv(Constants.REDIS_PORT)));
+            } else {
+                // we do not want to create a Jedis connection pool if idempotency is disabled
+                // null is ok as idempotency won't be called
+                this.jedisClient = null;
+            }
+        }
+    }
+
+    /**
+     * Set redis user and secret to connect to the redis server
```

Comment:
i'd suggest something like this - `set redis user and secret` makes me think it's mutating something 

```suggestion
     * Create jedis configuration to connect to the redis server
```

Reply:
This'll have to change with the no-passwords-in-env-vars-if-possible thing anyhow